core/drivers/stm32_i2c.c

```diff
@@ -692,11 +692,14 @@ int stm32_i2c_get_setup_from_fdt(void *fdt, int node,
 	memset(init, 0, sizeof(*init));
 
 	_fdt_fill_device_info(fdt, &info, node);
+	assert(info.reg != DT_INFO_INVALID_REG &&
+	       info.reg_size != DT_INFO_INVALID_REG_SIZE &&
+	       info.clock != DT_INFO_INVALID_CLOCK);
```

Comment:
Shouldn't `panic()` be used rather than `assert()`? (the latter is a no-op when `-DNDEBUG`).

Reply:
As previously done, we do not expect something wrong from the DT as it comes from the DT SoC level originally, that's why we use for most of our driver this check under assert only in debug mode.

Reply:
OK